### PR TITLE
Feature/authentication

### PR DESCRIPTION
**Pull request by modkaffes**
This adds support for frontend authentication and session management, connecting to auth server.

### CODE REVIEW

File: client/src/components/LogInForm.js [hunk 1]
```diff
@@ -1,57 +1,110 @@
...
+          required
+          autoComplete="off"
+          tabIndex="3"
+          aria-invalid="false"
+          aria-controls="password"
+          aria-label="Choose a strong password."
+          aria-activedescendant="password-item-0"
+        />
+        <label htmlFor="passphrase" className="d-none">
+          Passphrase
```
Comment (guylepage3) [inline client/src/components/LogInForm.js]: User's should not use a `Passphrase` in order to log in. We only reqire an email address and password to authenticate the user currently.

File: client/src/components/SignUpForm.js [hunk 2]
```diff
@@ -1,77 +1,109 @@
...
+          required
+          autoComplete="off"
+          tabIndex="3"
+          aria-invalid="false"
+          aria-controls="password"
+          aria-label="Choose a strong password."
+          aria-activedescendant="password-item-0"
+        />
+        <label htmlFor="passphrase" className="d-none">
+          Passphrase
```
Comment (guylepage3) [inline client/src/components/SignUpForm.js]: When the user signs up for Universe, they do not have to choose a `Passphrase`, we will authenticate the user with only their email address and password.

**Review by guylepage3 — CHANGES_REQUESTED** [review]

File: client/src/components/LogInForm.js (hunk 1 above)
Comment (modkaffes) [reply client/src/components/LogInForm.js]: Aha. Wasn’t actually using it for auth, it was there from before. I’m removing it. 

(Just email and password are used for logging in)
https://github.com/universelabs/universe-webapp/blob/65c352f62bafc809f122d7fa96f2711461019300/client/src/components/LogInForm.js#L11-L13

File: client/src/components/SignUpForm.js (hunk 2 above)
Comment (modkaffes) [reply client/src/components/SignUpForm.js]: Same as previous comment 👆 Will remove it

File: client/src/components/LogInForm.js (hunk 1 above)
Comment (guylepage3) [reply client/src/components/LogInForm.js]: It was actually not there from before. But yes. Let’s remove it please. 
Thank you sir.

https://github.com/universelabs/universe-webapp/blob/8900d6c9d228189bcb2babbf7be0dc38abc13b47/client/src/components/LogInForm.js

Comment (modkaffes) [reply client/src/components/LogInForm.js]: It seems I somehow picked it up from way back 😅
https://github.com/universelabs/universe-webapp/commit/d8fe10f813114c0f19927bbfad6fbbf8aa1b90ae#diff-92fc8f1f24c44cd5c75ca80953f8f5a0R59

Anw, it’s gone now 👍

File: client/src/components/SignUpForm.js [hunk 3]
```diff
@@ -1,77 +1,106 @@
...
+import { useDispatch } from 'redux-react-hook';
+import * as actions from '../constants/actions';
+import useForm from '../hooks/useForm';
 
-class SignUpForm extends Component {
-  render() {
-    return (
-      <form action="/signup" method="POST">
-        <div className="form-group">
-          <label htmlFor="firstName" className="d-none">
```
Comment (guylepage3) [inline client/src/components/SignUpForm.js:8]: Sorry. Did we change the perameters for this? Let's make sure we don't change things unless mentioned. Let's add the first name (optional) back into the sign up form. Thank you.

**Review by guylepage3 — CHANGES_REQUESTED** [review]
Let's add the `first name (optional)` back into the sign up form. Thank you.

File: client/src/components/SignUpForm.js (hunk 3 above)
Comment (guylepage3) [reply client/src/components/SignUpForm.js:8]: The form requirements should not have changed but somehow they did here as well..

Screenshot 2019-07-09 13 06 50
Screenshot 2019-07-09 13 07 02

**Comment by guylepage3** [comment]
Another issue that I have found is that the site is no longer secure. This is something that we will have to resolve as well @modkaffes @dantrevino 

image

**Comment by guylepage3** [comment]
Another issue.. When I go to the sign up page, then back to the login page and try to log in, nothing happens after I enter my credentials both correctly or incorrectly and click the "log in" button.

This is what I am seeing in my console... Never seen something like this before.

Screenshot 2019-07-09 13 28 56

**Comment by modkaffes** [comment]
> Another issue.. When I go to the sign up page, then back to the login page and try to log in, nothing happens after I enter my credentials both correctly or incorrectly and click the "log in" button.
> 
> This is what I am seeing in my console... Never seen something like this before.
> 
> Screenshot 2019-07-09 13 28 56

@guylepage3 thanks for the report, looking into it. 

What you’re seeing in your console is the output of [`redux-logger`](https://github.com/LogRocket/redux-logger)[middleware](https://github.com/universelabs/universe-webapp/pull/35/files#diff-36c47fb33022d2183169ad7c05522fadR5), used when developing. 

Edit: Resolved this offline.

File: client/src/components/SignUpForm.js (hunk 3 above)
Comment (modkaffes) [reply client/src/components/SignUpForm.js:8]: You’re right. Added this back.

**Comment by modkaffes** [comment]
> Another issue that I have found is that the site is no longer secure. This is something that we will have to resolve as well @modkaffes @dantrevino
> 
> image

This is only for local development. The deployed app is still going to be served over HTTPS. 
I can add config for running local dev env in HTTPS if need be.

**Pull request merged**